.github/workflows/integration_tests.yaml

```diff
@@ -22,69 +27,35 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: stable
-      - name: Build image
-        run: |
-          make image-tar
-      - name: Upload image
-        uses: actions/upload-artifact@v3
-        with:
-          name: image
-          path: build-output/ambassador-agent-image.tar
-  test:
-    runs-on: ubuntu-latest
-    needs: build-image
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-          ref: "${{ github.event.pull_request.head.sha }}"
-      - name: Get last run
-        uses: ./.github/actions/rerun-cache
-        id: get_last_run
-      - name: Setup Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: stable
       - name: Setup kubectl
         uses: azure/setup-kubectl@v3
         id: kubectl
-        if: steps.get_last_run.outputs.passed != 'success'
-      - name: Download image
-        uses: actions/download-artifact@v3
-        if: steps.get_last_run.outputs.passed != 'success'
-        with:
-          name: image
-      - name: Prepare kluster
-        uses: ./.github/actions/prepare-kluster
-        if: steps.get_last_run.outputs.passed != 'success'
+      - name: Provision Kubeception cluster
         id: kluster
-        with:
-          token: ${{ secrets.DEV_AMBASSADOR_AGENT_KUBECEPTION_TOKEN }}
-          image-tar: ambassador-agent-image.tar
-          github-sha: ${{ github.event.pull_request.head.sha }}
+        uses: datawire/infra-actions/provision-cluster@v0.2.6
+        with:
+          distribution: Kubeception
+          version: 1.26
+          kubeconfig: ${{ env.KUBECONFIG }}
+          kubeceptionToken: ${{ secrets.DEV_AMBASSADOR_AGENT_KUBECEPTION_TOKEN }}
+          gkeCredentials: '{"project_id": "foo"}' # See https://github.com/datawire/infra-actions/issues/66
+      - name: Add a private image registry to the cluster
+        run: make private-registry
+      - name: Build image
+        id: build
+        run: |
+          make push-image
+          echo "version=$(cat build-output/version.txt)" >> $GITHUB_OUTPUT
       - name: Run tests
-        if: steps.get_last_run.outputs.passed != 'success'
         uses: nick-invision/retry@v2
+        env:
+          A8R_AGENT_VERSION: ${{ steps.build.outputs.version }}
+          AMBASSADOR_AGENT_DOCKER_IMAGE: localhost:5000/ambassador-agent:${{ steps.build.outputs.version }}
+          KAT_SERVER_DOCKER_IMAGE: docker.io/datawiredev/kat-server:3.0.1-0.20220817135951-2cb28ef4f415
```

Comment:
random thought: 

I'm not too familiar with the Agent code base beyond knowing it was historically in emissary-ingress repo and just scrapes Edgissary for data and forwards it on to Ambassador Cloud.  But, I was surprised to see references to the `kat-server` so I will take it as a mental note to investigate this further during cooldown.